src/sql/workbench/browser/modelComponents/radioCardGroup.component.ts

```diff
@@ -0,0 +1,179 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import 'vs/css!./media/card';
+
+import {
+	Component, Inject, ChangeDetectorRef, forwardRef,
+	ElementRef, OnDestroy, Input, ViewChildren, QueryList
+} from '@angular/core';
+
+import { IComponent, IComponentDescriptor, IModelStore, ComponentEventType } from 'sql/workbench/browser/modelComponents/interfaces';
+import { ComponentBase } from 'sql/workbench/browser/modelComponents/componentBase';
+import * as azdata from 'azdata';
+import { getIconClass } from 'sql/workbench/browser/modelComponents/iconUtils';
+import { StandardKeyboardEvent } from 'vs/base/browser/keyboardEvent';
+import { KeyCode } from 'vs/base/common/keyCodes';
+import * as DOM from 'vs/base/browser/dom';
+
+@Component({
+	templateUrl: decodeURI(require.toUrl('./radioCardGroup.component.html'))
+
+})
+export default class RadioCardGroup extends ComponentBase implements IComponent, OnDestroy {
+	@Input() descriptor: IComponentDescriptor;
+	@Input() modelStore: IModelStore;
+	@ViewChildren('cardDiv') cardElements: QueryList<ElementRef>;
+
+	private selectedCard: azdata.RadioCard;
+	private focusedCard: azdata.RadioCard;
+	private iconClasses: { [key: string]: string } = {};
+
+	constructor(
+		@Inject(forwardRef(() => ChangeDetectorRef)) changeRef: ChangeDetectorRef,
+		@Inject(forwardRef(() => ElementRef)) el: ElementRef) {
+		super(changeRef, el);
+	}
+
+	ngOnInit(): void {
+		this.baseInit();
+	}
+
+	setLayout(layout: any): void {
+		this.layout();
+	}
+
+	ngOnDestroy(): void {
+		Object.keys(this.iconClasses).forEach((key) => {
+			DOM.removeCSSRulesContainingSelector(this.iconClasses[key]);
+		});
+		this.baseDestroy();
+	}
+
+	onKeyDown(event: KeyboardEvent): void {
+		if (!this.enabled || this.cards.length === 0) {
```

Comment:
may want to check for this.cards as well

Reply:
default value for this.cards is [], it should be safe to skip the null check for it.